plugins/query-resource-limit-group/src/main/java/org/opensearch/plugin/resource_limit_group/UpdateResourceLimitGroupRequest.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.resource_limit_group;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.cluster.metadata.ResourceLimitGroup;
+import org.opensearch.cluster.metadata.ResourceLimitGroup.ResourceLimit;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * A request for update Resource Limit Group
+ *
+ * @opensearch.internal
+ */
+public class UpdateResourceLimitGroupRequest extends ActionRequest implements Writeable.Reader<UpdateResourceLimitGroupRequest> {
+    String name;
+    List<ResourceLimit> resourceLimits;
+    String enforcement;
+    String updatedAt;
+
+    /**
+     * Default constructor for UpdateResourceLimitGroupRequest
+     * @param name - name for Resource Limit Group
+     */
+    public UpdateResourceLimitGroupRequest(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Constructor for UpdateResourceLimitGroupRequest
+     * @param resourceLimitGroup - A {@link ResourceLimitGroup} object
+     */
+    public UpdateResourceLimitGroupRequest(ResourceLimitGroup resourceLimitGroup) {
+        this.name = resourceLimitGroup.getName();
+        this.resourceLimits = resourceLimitGroup.getResourceLimits();
+        this.enforcement = resourceLimitGroup.getEnforcement();
+        this.updatedAt = resourceLimitGroup.getUpdatedAt();
+    }
+
+    /**
+     * Constructor for UpdateResourceLimitGroupRequest
+     * @param in - A {@link StreamInput} object
+     */
+    public UpdateResourceLimitGroupRequest(StreamInput in) throws IOException {
+        super(in);
+        name = in.readString();
+        if (in.readBoolean()) {
+            resourceLimits = in.readList(ResourceLimit::new);
+        }
+        enforcement = in.readOptionalString();
+        updatedAt = in.readString();
+    }
+
+    @Override
+    public UpdateResourceLimitGroupRequest read(StreamInput in) throws IOException {
+        return new UpdateResourceLimitGroupRequest(in);
+    }
+
+    /**
+     * Generate a UpdateResourceLimitGroupRequest from XContent
+     * @param parser - A {@link XContentParser} object
+     */
+    public static UpdateResourceLimitGroupRequest fromXContent(XContentParser parser) throws IOException {
+        ResourceLimitGroup resourceLimitGroup = ResourceLimitGroup.fromXContentOptionalFields(parser);
```

Comment:
I think this will only have the values coming from user input and rest of fields in ResourceLimitGroup object will be null. Is that okay ?

Reply:
Yes I created another parser in the resourcelimitgroup.java file that allows optional/null fields. You can see that here we're calling the fromXContentOptionalFields function instead of the fromXContent function